vllm/v1/worker/gpu_model_runner.py

```diff
@@ -1240,12 +1242,22 @@ def execute_model(
             intermediate_tensors = self.sync_and_slice_intermediate_tensors(
                 num_input_tokens, intermediate_tensors, True)
 
+        # Some attention backends only support CUDA graphs in pure decode.
+        # Assume cuda_graph_supported is false if it does not exist.
+        attention_cuda_graphs = all(
+            getattr(m, "cuda_graph_supported", False)
+            for _, m in attn_metadata.items())
```

Comment:
this can be expensive, please calculate it during init and save the result to use for every forward pass.

Reply:
oh this is per-batch state, so we cannot calculate it during init.

can we just query the first layer? and leave a comment saying that we only support homogeneous attention layers right now.

I'm afraid checking all layers would be too expensive and make the full-cudagraph benefit meaningless.

Reply:
It doesn't seem bad based on perf results, but I think I could do that. Or we could save it on the builder

Reply:
@youkaichao I thought about this more and I think given we already construct metadata per-layer, this seems like it would be negligible overhead. Also no need to introduce complexity/a restriction we might need to remove later anyway.

Reply:
I think that's the right approach, will update